File: src/main/scala/ch/cern/ExecutorPluginScala/DemoPlugin.scala
package ch.cern.ExecutorPluginScala

// Basic example of Spark Executor Plugin in Scala
// code adapted from [SPARK-24918]

class DemoPlugin extends org.apache.spark.ExecutorPlugin {

  override def init() : Unit = {
    DemoPlugin.numSuccessfulPlugins += 1
  }

  override def shutdown() : Unit = {
    DemoPlugin.numSuccessfulTerminations += 1
  }
}

object DemoPlugin {
  var numSuccessfulPlugins : Int = 0
  var numSuccessfulTerminations: Int = 0
}

File: src/main/java/ch/cern/ExecutorPluginJava/DemoPlugin.java
package ch.cern.ExecutorPluginJava;

// Basic example of Spark Executor Plugin in Java
// This just shows how to write a plugin, it does not do any action besides incrementing a cuonter
// code from [SPARK-24918]

public class DemoPlugin implements org.apache.spark.ExecutorPlugin {

    public static int numSuccessfulPlugins = 0;
    public static int numSuccessfulTerminations = 0;

    public void init() {
        numSuccessfulPlugins++;
    }
    public void shutdown() {
        numSuccessfulTerminations++;
    }
}
